include/up-cpp/uri/tools/IpAddress.h

```diff
@@ -19,13 +19,24 @@
  * under the License.
  * 
  * SPDX-FileType: SOURCE
- * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
  * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef IP_ADDRESS_H_
 #define IP_ADDRESS_H_
 
-#include <iostream> 
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+// Forward declare UAuthority so we can implement a constructor that consumes it
```

Comment:
No. This is a forward declaration with no definition. This is a C/C++ism for _"this type will be defined elsewhere, but I'm going to make a passing reference to it here"_ and the compiler won't be bothered by it. It avoids `#include`'ing a file here that isn't otherwise needed.

I have compiled and tested this code, after all :grin: